### ISSUE TITLE
M1 Mac Error

### ISSUE THREAD

**Issue by japr** [ReSwift/ReSwift#468]
Hi, I just tried to compile a small project where I use ReSwift on a new Mac with M1 chip, but I'm getting the following error when trying to build the project:

![Screen Shot 2020-12-28 at 10 43 55 PM](https://user-images.githubusercontent.com/4094061/103257377-f882b080-495e-11eb-9157-3ad11e88129c.png)
Has anyone experienced the same error? It seems that a new build settings value should be added when creating the framework, but I'm not sure.

Any help or comment is appreciated.

Thanks!

**Comment by mjarvis** [comment]
Looks like we'll need to confirm / add support for the M1 chip's target architecture. Thanks for the ticket.

**Comment by mjarvis** [comment]
@japr How are you integrating ReSwift into your project? Swift Package Manager, CocoaPods, Carthage, or manually?

**Comment by japr** [comment]
I'm using cocoapods

**Comment by mjarvis** [comment]
Thanks @japr . I don't currently have an M1 mac but will try to reach out to others in order to investigate.

**Comment by thattyson** [comment]
@japr I spent about an hour hacking Xcode settings to solve this issue for a different cocoapods dependency. I’m not 100% sure which of these solved it, but in order of what I think was most likely the solution:

1. Set `Build Active Architectures Only` to `No` in **Pods Project** Build Settings (for me I had to remove a post-install script from my podfile to stop setting it to `Yes`)
2. Add `x86_64` and `x86` to `Architectures` in **Your Project** Build Settings.
3. A previous solution to an M1 cocoapod issues was to add `config.build_settings["EXCLUDED_ARCHS[sdk=iphonesimulator*]"] = "arm64"` to my podfile post-install. I've removed this and looks like an Xcode or Cocoapods update have solved the issue this fixed.

I'm on Xcode 12.3, Cocoapods 1.10.1, Big Sur 11.1. This fixed the issue for my M1 and ARM computers. Good luck!

Here's a link to the Apple Developer Answer that I found: https://developer.apple.com/forums/thread/123614?answerId=385588022#385588022
And to SO answer where I tried Active Architectures = No: https://stackoverflow.com/a/58230124/3128245
